addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/handler/AirQualityHandler.java

```diff
@@ -0,0 +1,308 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.airquality.handler;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Calendar;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.airquality.AirQualityBindingConstants;
+import org.openhab.binding.airquality.internal.AirQualityConfiguration;
+import org.openhab.binding.airquality.internal.json.AirQualityJsonResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link AirQualityHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Kuba Wolanin - Initial contribution
+ * @author Łukasz Dywicki
+ */
+public class AirQualityHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(AirQualityHandler.class);
+
+    private static final String URL = "http://api.waqi.info/feed/geo:%LOCATION%/?token=%APIKEY%";
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private ScheduledFuture<?> refreshJob;
+
+    AirQualityJsonResponse aqiResponse;
+
+    private Gson gson;
+
+    public AirQualityHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Air Quality handler.");
+        super.initialize();
+
+        AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+        logger.debug("config location = {}", config.getLocation());
+        logger.debug("config refresh = {}", config.getRefresh());
```

Comment:
Hey! You've removed the line, which is totally fine. What I actually meant to suggest was to output a hardcoded "config apikey = (omitted from logging)". This way the user still knows/has the feeling his configuration was taken into account. I believe that is the actual purpose of these logger lines!?
This is a mere unimportant detail. Do as you feel it's right ;)